### DISCUSSION TITLE
Renaming Main Menu Items

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9626] [Q&A]
Hello - I am interested in renaming main menu items (such as: File, Open Workspace, Open Recent Workspace, etc.) and I wanted to know how I could go about finding the location of the labels for these? I know how to hide menu items but am not necessarily sure where to find the labels. Thanks!

**Comment by vince-fugnitto** [comment]
@sboudreaux96 there is no API as far as I am aware to update an existing menu item (ex: its `label`). Perhaps it is something we can add to the framework to make it easier for downstream applications.

I was able to get it to work by doing the following:

```ts
registerMenus(menus: MenuModelRegistry): void {

    // Collect existing `File` menu items.
    const fileMenus = menus.getMenu(CommonMenus.FILE).children;

    // Remove existing `File` menu.
    menus.unregisterMenuAction(CommonMenus.FILE.slice(-1)[0]);

    // Register updated `File` menu and children.
    menus.registerSubmenu(CommonMenus.FILE, 'Custom File');
    for (const menu of fileMenus) {
        menus.registerMenuNode(CommonMenus.FILE, menu);
    }

}
```


https://user-images.githubusercontent.com/40359487/122949164-71305b80-d349-11eb-9579-5e92c842942f.mp4
